libp2p/dialer.nim

```diff
@@ -150,7 +150,8 @@ proc dialAndUpgrade(
         if not isNil(result):
           return result
 
-proc tryReusingConnection(self: Dialer, peerId: PeerId): Future[Opt[Muxer]] {.async.} =
+proc tryReusingConnection(self: Dialer, peerIdOpt: Opt[PeerId]): Opt[Muxer] =
```

Comment:
I think if we wanna try reusing a connection, a peer id should always be passed. If it's none, don't call the proc.

Reply:
Yeah this was just to make it easier at callsite, tried to improve